packages/block-editor/src/hooks/background.js

```diff
@@ -0,0 +1,271 @@
+ * attributes.
+ *
+ * @param {Object} props Block props.
+ * @return {boolean}     Whether or not the block has a background image value set.
+ */
+export function hasBackgroundImageValue( props ) {
+	const hasValue =
+		!! props.attributes.style?.background?.backgroundImage?.id ||
+		!! props.attributes.style?.background?.backgroundImage?.url;
+
+	return hasValue;
+}
+
+/**
+ * Determine whether there is block support for background.
+ *
+ * @param {string} blockName Block name.
+ * @param {string} feature   Background image feature to check for.
+ *
+ * @return {boolean} Whether there is support.
+ */
+export function hasBackgroundSupport( blockName, feature = 'any' ) {
+	if ( Platform.OS !== 'web' ) {
+		return false;
+	}
+
+	const support = getBlockSupport( blockName, BACKGROUND_SUPPORT_KEY );
+
+	if ( support === true ) {
+		return true;
+	}
+
+	return !! support?.[ feature ];
+}
+
+/**
+ * Resets the background image block support attributes. This can be used when disabling
+ * the background image controls for a block via a `ToolsPanel`.
+ *
+ * @param {Object} props               Block props.
+ * @param {Object} props.attributes    Block's attributes.
+ * @param {Object} props.setAttributes Function to set block's attributes.
+ */
+export function resetBackgroundImage( { attributes = {}, setAttributes } ) {
+	const { style = {} } = attributes;
+
+	setAttributes( {
+		style: cleanEmptyObject( {
+			...style,
+			background: {
+				...style?.background,
+				backgroundImage: undefined,
+			},
+		} ),
+	} );
+}
+
+function InspectorImagePreview( { url: imgUrl } ) {
+	const imgLabel = getFilename( imgUrl );
+	return (
+		<ItemGroup as="span">
+			<HStack justify="flex-start" as="span">
+				<img src={ imgUrl } alt="" />
+				<FlexItem as="span">
+					<Truncate
+						numberOfLines={ 1 }
+						className="block-editor-hooks__background__inspector-media-replace-title"
+					>
+						{ imgLabel }
+					</Truncate>
+				</FlexItem>
+			</HStack>
+		</ItemGroup>
+	);
+}
+
+function BackgroundImagePanelItem( props ) {
+	const { attributes, clientId, setAttributes } = props;
+
+	const { id, url } = attributes.style?.background?.backgroundImage || {};
+
+	const { mediaUpload } = useSelect( ( select ) => {
+		return {
+			mediaUpload: select( blockEditorStore ).getSettings().mediaUpload,
+		};
+	} );
+
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const onUploadError = ( message ) => {
+		createErrorNotice( message, { type: 'snackbar' } );
+	};
+
+	const onSelectMedia = ( media ) => {
+		if ( ! media || ! media.url ) {
+			const newStyle = {
+				...attributes.style,
+				background: {
+					...attributes.style?.background,
+					backgroundImage: undefined,
+				},
+			};
+
+			const newAttributes = {
+				style: cleanEmptyObject( newStyle ),
+			};
+
+			setAttributes( newAttributes );
+			return;
+		}
+
+		if ( isBlobURL( media.url ) ) {
+			// TODO: Might need to revoke the blob URL.
```

Comment:
what does this mean?

Reply:
Oh, thanks for the reminder! I needed to double check — when media is uploaded a temporary blob URL is created and added to a cache. It turns out it looks like the revoking / clearing of the blob URL is already handled for us (if I'm following correctly) by the upload logic [here](https://github.com/WordPress/gutenberg/blob/53a44bedfc90451592c895e245bc662f3b345333/packages/media-utils/src/utils/upload-media.js#L69) once a file has been uploaded, so we don't need to handle it in this component. I've removed the TODO comment.